src/plugins/workspace/server/workspace_client.ts

```diff
@@ -166,9 +166,12 @@ export class WorkspaceClientWithSavedObject implements IWorkspaceDBImpl {
           throw new Error(DUPLICATE_WORKSPACE_NAME_ERROR);
         }
       }
-      await client.update<Omit<WorkspaceAttribute, 'id'>>(WORKSPACE_TYPE, id, attributes, {
+      await client.create<Omit<WorkspaceAttribute, 'id'>>(WORKSPACE_TYPE, attributes, {
```

Comment:
Q: what it calling `update()` with a non-existing `id`? will it create a new workspace with the id?

Reply:
The `client.create` will create a new saved object if the id doesn't exists. What do you think about we check if the id exists first?

Reply:
> Q: what it calling `update()` with a non-existing `id`? will it create a new workspace with the id?

Yes, that will create a new workspace.  we could add version into `SavedObjectsCreateOptions` to have `[Optimistic concurrency control]`, if version is not match that will not perform create/update 

```
 await client.create<Omit<WorkspaceAttribute, 'id'>>(WORKSPACE_TYPE, attributes, {
        id,
        permissions,
        overwrite: true,
        version: workspaceInDB.version,
      });
```

Reply:
@Hailong-am Is this version info already exists? Will it be updated automatically?

Reply:
Yes, `get` method will return version info. https://github.com/ruanyl/OpenSearch-Dashboards/blob/workspace/src/core/server/saved_objects/service/lib/repository.ts#L1095, it's a encoded value based on `_seq_no` and `_primary_term` which are used for [optimistic concurrency control](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/docs-index_.html#optimistic-concurrency-control-index)

Reply:
@Hailong-am @ruanyl The workspace update API will be rejected if non-exists id passed. There was a `client.get` before we call `client.create`.  The `client.get` will throw error if the id wasn't exists.

Reply:
> The workspace update API will be rejected if non-exists id passed. There was a client.get before we call client.create. The client.get will throw error if the id wasn't exists.

Then I guess we're good to create with overwrite